packages/dds/tree/src/test/cursorTestSuite.ts

```diff
@@ -184,17 +179,63 @@ export interface SpecialCaseBuilder<TData> {
  * @param dataFromCursor - Constructs a `TData` from the provided cursor (which might not be a `TCursor`).
  * @param testData - A collection of test cases to evaluate the cursor with. Actual content of the tree is only validated if a `reference` is provided:
  * otherwise only basic traversal and API consistency will be checked.
+ *
+ * @typeParam TData - Format which the cursor reads. Must be JSON compatible.
+ * @typeParam TCursor - Type of the cursor being tested.
+ */
+export function testSpecializedCursor<TData, TCursor extends ITreeCursor>(config: {
+    cursorName: string;
+    builders: SpecialCaseBuilder<TData>;
+    cursorFactory: (data: TData) => TCursor;
+    dataFromCursor: (cursor: ITreeCursor) => TData;
+    testData: readonly { name: string; data: TData; reference?: JsonableTree }[];
+}): Mocha.Suite {
+    return testTreeCursor(config);
```

Comment:
What's the purpose of wrapping this? The only difference I see is that dataFromCursor isn't optional here but I also don't see testTreeCursor being called elsewhere.

Reply:
There are two exported entry points to this test suite:
1. testGeneralPurposeTreeCursor: this is for cursors which can be constructed from any jsonable tree.
2. testSpecializedCursor: this is for cursors which can't handle all possible trees, for example JsonCursor and mt upcoming uniform chunk cursor.